File: Alfonso Equation Editor/Preferences/Preference Panes/SearchFieldEnabledValueTransformer/SearchFieldEnabledValueTransformer.m
//
//  SearchFieldEnabledValueTransformer.m
//  Brian Cliette Uber Course
//
//  Created by Alfonso Maria Tesauro on 03/04/16.
//  Copyright © 2016 Alfonso Maria Tesauro. All rights reserved.
//

#import "SearchFieldEnabledValueTransformer.h"

@implementation SearchFieldEnabledValueTransformer

+ (Class)transformedValueClass {
    return [NSColor class];
}

+ (BOOL)allowsReverseTransformation {
    return NO;
}

- (id)transformedValue:(id)value {
    
    if (value == nil) return nil;
    
    if (![value isKindOfClass:[NSNumber class]]) return nil;
    
    if ([value boolValue]) {
        return [NSColor grayColor];
    }
    
    return [NSColor whiteColor];
}

@end

File: Alfonso Equation Editor/Preferences/Preference Panes/SearchFieldEnabledValueTransformer/SearchFieldEnabledValueTransformer.swift
//
//  SearchFieldEnabledValueTransformer.swift
//  Brian Cliette Uber Course
//
//  Created by Alfonso Maria Tesauro on 03/04/16.
//  Copyright © 2016 Alfonso Maria Tesauro. All rights reserved.
//

import Cocoa

class SearchFieldEnabledValueTransformer: ValueTransformer {
    
    override class func transformedValueClass() -> AnyClass {
        return NSColor.self
    }
    
    override class func allowsReverseTransformation() -> Bool {
        return false
    }
    
    override func transformedValue(_ value: Any?) -> Any? {
        guard let value = value else { return nil }
        
        guard let number = value as? NSNumber else { return nil }
        
        if number.boolValue {
            return NSColor.gray
        }
        
        return NSColor.white
    }
}
